.circleci/config.yml

```diff
@@ -102,23 +102,33 @@ jobs:
               if [[ -f "$subdir"/test_requirements.txt ]]; then
                 pip install -r "$subdir"/test_requirements.txt
               fi
+              if [[ -f "$subdir"/requirements_dev.txt ]]; then
+                pip install -r "$subdir"/requirements_dev.txt
+              fi
               done
-            pip install cython wheel
-            pip install .
+            pip install cython wheel cgen
+            pip install .[all]
             dwave install --all --yes
 
       - run:
-          name: run individual package unittests
+          name: run individual package unit tests
           command: |
             . env/bin/activate
             for subdir in */; do
               if [[ -d "$subdir"/tests ]] && [[ "$subdir" != "env/" ]]; then
                 cd "$subdir"
                 pwd
                 if [[ -f "setup.py" ]]; then
+                  if [[ "$subdir" == "dwave-gate/" ]]; then
+                    python dwave/gate/simulator/operation_generation.py
+                  fi
                   python setup.py build_ext --inplace 2>/dev/null
                 fi
-                python -m unittest discover
+                if [[ -f "requirements_dev.txt" ]] && grep -q pytest "requirements_dev.txt"; then
+                  python -m pytest tests/
+                else
+                  python -m unittest discover
+                fi
```

Comment:
Perhaps not the most elegant solution, but it works. Could also just use pytest for dwave-gate (and unittest for all others), or check for pytest in e.g., `pyproject.toml` and add it as an install-requires line there since this relies on there being a `requirements_dev.txt` file with `pytest` added as a dependency.